mavros/src/plugins/setpoint_raw.cpp

```diff
@@ -217,6 +238,47 @@ class SetpointRawPlugin : public plugin::PluginBase,
 				body_rate,
 				req->thrust);
 	}
+
+	void rpyt_cb(const mav_msgs::RollPitchYawrateThrustConstPtr msg)
+	{
+	
+		if(ignore_rpyt_messages_){
+			ROS_FATAL("Recieved roll_pitch_yaw_thrust_rate message, but ignore_rpyt_messages_ is true: "
+	 	        "the most likely cause of this is a failure to specify the thrust_scaling_factor,	"
+			"yaw_rate_scaling_factor or system_mass_kg parameters");
+		     	return;
+		}
+
+		//the masks are much more limited than the docs would suggest so we don't use them
+		uint8_t type_mask = 0;
+
+		geometry_msgs::Quaternion orientation = tf::createQuaternionMsgFromRollPitchYaw( msg->roll, msg->pitch, 0);
+		
+		double thrust = std::min(1.0, std::max(0.0, msg->thrust.z * thrust_scaling_ * system_mass_kg_));
```

Comment:
As @mhkabir says, this should be somehow documented in a commit above this line, in order to understand how this actually works. The same thing goes for https://github.com/mavlink/mavros/pull/964/files#diff-57d2f68a3e0e5cfa2db14ac20d8845ddR110.